### ISSUE TITLE
Rewind support

### ISSUE THREAD

**Issue by cgillum** [microsoft/durabletask-dotnet#21]
Durable Functions and the Azure Storage backend both support orchestration rewind, which allows restoring an orchestration instance back to its previous state after it has failed. This functionality should be made available to this SDK.

Rewind is currently in preview status. As part of this work, we should bring it onto the path to GA. This would likely require some changes to DTFx, such as completing the following work item: https://github.com/Azure/durabletask/issues/731.

### Checklist

- [ ] Core DTFx updates
- [ ] New DF API surface area
- [ ] gRPC/Protobuf contract support
- [ ] Integration tests

**Comment by jviau** [comment]
Rolling this into Core DTFx and GA'ing does look possible. However, I wonder if we this is "feature-parity" for Durable Functions though? Is it possible to make a more targeted change for DF isolated to support rewind on just the known supported backends.

**Comment by cgillum** [comment]
Yes, I think I was being a little overzealous by adding the "Core DTFx updates" checkbox. That probably should have been a separate work item in the Azure/durabletask repo. Feature parity can be achieved by simply plugging into the existing implementation as you suggested.

**Comment by source-studio** [comment]
May I ask if there is an update on when this feature will be implemented?

**Comment by YunchuWang** [comment]
@sophiatev for update on this and release timeline

**Comment by sophiatev** [comment]
This feature has since been released, but I need to sync with @torosent as to whether or not we plan to officially GA this.

I'm also not sure if we plan on doing the changes proposed in the linked durabletask issue. Rewind has since been implemented in the SDK layer, but it still requires backend-specific-logic to handle it correctly (which so far has only been implemented in DTS).

**Comment by BOLT04** [comment]
@sophiatev I'm wondering if there are any plans for this being GA, does anyone know? Yes, it's 2026 and this feature would be great to have 😄

**Comment by sophiatev** [comment]
@BOLT04 if you mean rewind in this SDK, I think we can safely call it "GA". Found [here](https://github.com/microsoft/durabletask-dotnet/blob/b721667211d1c2342e1fcc41e6bd941916f4728c/src/Client/Grpc/GrpcDurableTaskClient.cs#L547).

Note that this currently only works in the Functions use-case.

**Comment by BOLT04** [comment]
> [@BOLT04](https://github.com/BOLT04) if you mean rewind in this SDK, I think we can safely call it "GA". Found [here](https://github.com/microsoft/durabletask-dotnet/blob/b721667211d1c2342e1fcc41e6bd941916f4728c/src/Client/Grpc/GrpcDurableTaskClient.cs#L547).
> 
> Note that this currently only works in the Functions use-case.

@sophiatev sorry that is not the client we use, we use `DurableTaskClient`. It's not supported as we can see here:

https://github.com/microsoft/durabletask-dotnet/blob/b721667211d1c2342e1fcc41e6bd941916f4728c/src/Client/Core/DurableTaskClient.cs#L491-L495

**Comment by sophiatev** [comment]
@BOLT04 that class is an abstract class with no method implementations. Note that this exception is thrown for all methods in that class.

It is very likely you are using the [`GrpcDurableTaskClient`](https://github.com/microsoft/durabletask-dotnet/blob/b721667211d1c2342e1fcc41e6bd941916f4728c/src/Client/Grpc/GrpcDurableTaskClient.cs#L547) implementation of it, which has the method implemented as seen in the link, unless you are explicitly creating a [`ShimDurableTaskClient`](https://github.com/microsoft/durabletask-dotnet/blob/main/src/Client/OrchestrationServiceClientShim/ShimDurableTaskClient.cs#L28) to interact directly with the backend.
